src/common/snippets/src/lowered/pass/insert_broadcastmove.cpp

```diff
@@ -0,0 +1,86 @@
+// Copyright (C) 2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "snippets/lowered/pass/insert_broadcastmove.hpp"
+#include "snippets/utils.hpp"
+#include "snippets/lowered/linear_ir.hpp"
+#include "snippets/snippets_isa.hpp"
+#include "snippets/itt.hpp"
+
+namespace ov {
+namespace snippets {
+namespace lowered {
+namespace pass {
+
+bool InsertBroadcastMove::run(LinearIR& linear_ir) {
+    OV_ITT_SCOPED_TASK(ov::pass::itt::domains::SnippetsTransform, "Snippets::InsertBroadcastMove")
+    bool modified = false;
+
+    auto supports_broadcasting = [](const std::shared_ptr<ov::Node>& n) {
+      return ov::op::util::supports_auto_broadcast(n) ||
+             n->get_autob().m_type == ov::op::AutoBroadcastType::NUMPY ||
+             is_type<ov::op::v0::PRelu>(n);
+    };
+    auto dont_need_broadcasting = [](const ov::Output<ov::Node>& v){
+        // We don't need to insert BroadcastMove after the following operations:
+        // - Scalar has emitter with explicit broadcasting
+        // - VectorBuffer has scalar output shape to avoid broadcast conflicts and manually shape insertion.
+        // - Fill can be inserted only after VectorBuffer, and should be ignored as well.
+        return utils::is_scalar_constant(v.get_node_shared_ptr()) ||
+               ov::is_type<ov::snippets::op::VectorBuffer>(v.get_node_shared_ptr()) ||
+                ov::is_type<ov::snippets::op::Fill>(v.get_node_shared_ptr());
```

Comment:
```suggestion
               ov::is_type<ov::snippets::op::Fill>(v.get_node_shared_ptr());
```

I cannot skip this moment :smile:

Reply:
omg, it's a felony 😆 
corrected